docs/ibc/light-clients/client-state.md

```diff
@@ -1,3 +1,65 @@
 <!--
 order: 2
--->
\ No newline at end of file
+-->
+
+# Implementing the ClientState interface
+
+Learn how to implement the [Client State](https://github.com/cosmos/ibc-go/blob/main/modules/core/exported/client.go#L36) interface.
+
+### ClientType
+
+`ClientType` should return a unique string identifier of the light client.
+
+### GetLatestHeight
+
+`GetLatestHeight` should return the latest block height.
+
+### Validate
+
+`Validate` should validate every client state field and should return an error if any value is invalid. The light client
+implementor is in charge of determining which checks are required. See the [tendermint light client implementation](https://github.com/cosmos/ibc-go/blob/main/modules/light-clients/07-tendermint/client_state.go#L111)
+as a reference.
+
+### Status
+
+`Status` must return the status of the client. Only `Active` clients are allowed to process packets. All
+possible Status types can be found [here](https://github.com/cosmos/ibc-go/blob/main/modules/core/exported/client.go).
```

Comment:
It might be useful to explain what `Frozen` and `Expired` mean

Reply:
A bulleted list could be nice, giving a short description of each of the statuses.

Reply:
I looked around at our usages of the statues, the only one the connection keeper cares about is [`Active`](https://github.com/cosmos/ibc-go/blob/fb9dedd706e5835cdd607392320df79fcea7ca7f/modules/core/03-connection/keeper/verify.go#L172).  [`Frozen`](https://github.com/cosmos/ibc-go/blob/c86d27fc280cfb342a9e4689b381e5823441b694/modules/light-clients/07-tendermint/proposal_handle.go#L40) is used in our implementation of `CheckSubstituteAndUpdateState`, however this is an interface method and so will look different in custom light clients. I didn't see any usage of `Expired`. My take away is that use of `Frozen`/`Expired` are for internal implementation of the client, but it won't have an impact in the same way `Active` does.

Please let me know if my understanding is incorrect!

Reply:
Frozen, Expired were previously required in the `CheckSubstituteAndUpdateState` logic that has now been removed. They are useful in determining why a client is not active. A client is `Frozen` if misbehaviour has successfully been submitted against it. Expired clients are those whose latest consensus state update is too old to process a new update given the client state parameters. In the tendermint case, it is specifically in relation to the trusting period, that is, the next update must be before the lastest consensus state time + the trusting period

I think it is still useful to explain these concepts even if core IBC only cares if the status is Active as it'll be useful for UX purposes

Reply:
Here's some development context on how this function came to be, see issue #98 and this [thread](https://github.com/cosmos/ibc-go/pull/140#discussion_r622372667)z

Reading back, maybe we can mention that this is also used for the Status gRPC route?